File: CS/FilterLookupListView.Module/BusinessObjects/Product.cs
using System;
using System.ComponentModel;

using DevExpress.Xpo;
using DevExpress.Data.Filtering;

using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Validation;

namespace FilterLookupListView.Module.BusinessObjects {
   //[DefaultClassOptions]
   public class Product : BaseObject {
      public Product(Session session) : base(session) {}
      public override void AfterConstruction() {
         base.AfterConstruction();
      }
      private String productName;
      public String ProductName {
         get {
            return productName;
         }
         set {
            SetPropertyValue("ProductName", ref productName, value);
         }
      }
      [Association("P-To-C")]
      public XPCollection<Accessory> Accessories {
         get {
            return GetCollection<Accessory>("Accessories");
         }
      }
   }

}

File: VB/FilterLookupListView.Module/BusinessObjects/Product.vb
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Xpo

Namespace FilterLookupListView.Module.BusinessObjects
	Public Class Product
		Inherits BaseObject

		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Private _productName As String
		Public Property ProductName() As String
			Get
				Return _productName
			End Get
			Set(ByVal value As String)
				SetPropertyValue(NameOf(ProductName), _productName, value)
			End Set
		End Property
		<Association("P-To-C")>
		Public ReadOnly Property Accessories() As XPCollection(Of Accessory)
			Get
				Return GetCollection(Of Accessory)("Accessories")
			End Get
		End Property
	End Class
End Namespace
